src/diagnostics_component/utils/file_info_utils/src/file_info_utils.c

```diff
@@ -168,13 +168,11 @@ bool FileInfoUtils_FillFileInfoWithNewestFilesInDir(FileInfo* logFiles, size_t l
 
     if (!succeeded)
     {
+        // Free all the file names, if allocated.
         for (int i = 0; i < logFileSize; ++i)
         {
-            FileInfo logFileEntry = logFiles[i];
-
-            free(logFileEntry.fileName);
-
-            memset(&logFileEntry, 0, sizeof(FileInfo));
+            free(logFiles[i].fileName);
+            memset(&logFiles[i], 0, sizeof(FileInfo));
```

Comment:
nit: no need to memset on each iteration.  After this loop, just call memset like done on L107.